src/commands/synthetics/batch.ts

```diff
@@ -57,14 +58,16 @@ export const waitForBatchToFinish = async (
       reporter
     )
 
-    reportResults(resultsToReport, pollResultMap, resultDisplayInfo, hasBatchExceededMaxPollingDate, reporter)
+    reportResults(resultsToReport, pollResultMap, resultDisplayInfo, safeDeadlineReached, reporter)
 
     oldIncompleteResultIds = incompleteResultIds
 
+    if (safeDeadlineReached) {
+      throw new CriticalError('SAFE_DEADLINE_REACHED', 'Datadog did not answer on time.')
```

Comment:
This error message seems a bit misleading: the backend was responding, and it's the batch that was not timing out whereas it was expected to, isn't it?
I would rephrase that with a unified error message, such as "the batch didn't timeout after the expected timeout period".